File: Easy/Eat Twice (EATTWICE)/eat.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
	int t;
	cin>>t;
	while(t-->0)
	{
	    int n,m;
	    cin>>n>>m;
	    int arr[m]={0};
	    int d,v;
	    for(int i=0;i<n;i++)
	    {
	        cin>>d>>v;
	        if(arr[d-1]<v)
	        {
	            arr[d-1]=v;
	        }
	    }
	    sort(arr,arr+m);
	    long long tot=arr[m-2]+arr[m-1];
	    cout<<tot<<endl;
	}
	return 0;
}

File: Easy/Eat Twice (EATTWICE)/eat.py
t = int(input())
while t > 0:
    t -= 1
    tmpText1 = input()
    n = int(tmpText1.split(" ")[0])
    m = int(tmpText1.split(" ")[1])
    arr = [0 for i in range(m)]
    for i in range(n):
        tmpText2 = input()
        d = int(tmpText2.split(" ")[0])
        v = int(tmpText2.split(" ")[1])
        if arr[d-1] < v:
            arr[d-1] = v
    arr.sort()
    tot = arr[m-2] + arr[m-1]
    print(tot)
